tools/generator/Method.cs

```diff
@@ -456,7 +457,20 @@ string GetDelegateType ()
 			}
 			return delegate_type;
 		}
-		
+
+		public bool IsCovariantOverride (Method overriddenMethod)
+		{
+			//A method with a covariant return type would have the following to be true:
+			// 1) Neither is generic
+			// 2) JavaName is the same
+			// 3) retval.JavaName is *different*
+			// 4) JavaSignature for parameters is the same
+			return !IsGeneric && !overriddenMethod.IsGeneric &&
+				JavaName == overriddenMethod.JavaName && 
+				retval.JavaName != overriddenMethod.retval.JavaName && 
+				Parameters.JavaSignature == overriddenMethod.Parameters.JavaSignature;
```

Comment:
This should check for `Parameters.Count` equality before hitting `JavaSignature`, to save on extra work.